Kernel/Devices/USB/USBTransfer.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2021 Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Devices/USB/USBTransfer.h>
+#include <Kernel/VM/ContiguousVMObject.h>
+#include <Kernel/VM/MemoryManager.h>
+
+namespace Kernel::USB {
+
+RefPtr<USBTransfer> USBTransfer::create_transfer(const USBPipe& pipe, u16 len)
+{
+    return adopt(*new USBTransfer(pipe, len));
+}
+
+USBTransfer::USBTransfer(const USBPipe& pipe, u16 len)
+    : m_pipe(pipe)
+    , m_transfer_data_size(len)
+{
+    // Initialize data buffer for transfer
+    // This will definitely need to be refactored in the future, I doubt this will scale well...
+    auto vmobj = ContiguousVMObject::create_with_size(PAGE_SIZE);
+    m_data_buffer = MemoryManager::the().allocate_kernel_region_with_vmobject(*vmobj, PAGE_SIZE, "USB Transfer Buffer", Region::Access::Read | Region::Access::Write);
+}
+
+USBTransfer::~USBTransfer()
+{
+    m_data_buffer.clear();
```

Comment:
Uhhh, this _does_ free up the allocated region... Right....?

Reply:
This will automatically be deconstructed once the `USBTransfer` leaves scope right? Do I need to do any manual cleanup of `m_data_buffer` to release the `Region` allocated here?

Reply:
OwnPtr will call the stored region's destructor automatically